internal/module/statusupdater.go

```diff
@@ -0,0 +1,53 @@
+package module
+
+import (
+	"context"
+	"fmt"
+
+	ootov1alpha1 "github.com/qbarrand/oot-operator/api/v1alpha1"
+	"github.com/qbarrand/oot-operator/internal/daemonset"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -source=statusupdater.go -package=module -destination=mock_statusupdater.go
+
+type StatusUpdater interface {
+	UpdateModuleStatus(ctx context.Context, mod *ootov1alpha1.Module, nodes []v1.Node) error
+}
+
+type statusUpdater struct {
+	client    client.Client
+	daemonAPI daemonset.DaemonSetCreator
+}
+
+func NewStatusUpdater(client client.Client, daemonAPI daemonset.DaemonSetCreator) StatusUpdater {
+	return &statusUpdater{
+		client:    client,
+		daemonAPI: daemonAPI,
+	}
+}
+
+func (s *statusUpdater) UpdateModuleStatus(ctx context.Context, mod *ootov1alpha1.Module, nodes []v1.Node) error {
+	dsList, err := s.daemonAPI.ModuleDaemonSets(ctx, mod.Name, mod.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get all module's daemonsets: %w", err)
+	}
+	numDesired := int32(len(nodes))
```

Comment:
we can't use DaemonSets for number of desired, since it is possible that when we try to set the status, some or all of the daemonsets are not running yet. That's why i am using nodes. Nodes represent all the nodes that driver-container (and device-plugin) should be running on